Cargo.toml

```diff
@@ -21,8 +21,16 @@ categories = ["network-programming", "data-structures"]
 
 [dependencies]
 byteorder = "1.0.0"
-iovec = "0.1"
 serde = { version = "1.0", optional = true }
 
+[dependencies.iovec]
+version = "0.1"
+optional = true
+
 [dev-dependencies]
 serde_test = "1.0"
+
+[features]
+default = ["use_std"]
+use_alloc = []
+use_std = ["iovec"]
```

Comment:
Err, other way around: `use_alloc` and `use_std` are orthogonal. `use_alloc` is intended for `no_std` use, i.e. `--no-default-features --features=use_alloc`

bytes builds without it per this PR, but with so many things disabled it isn't particularly useful